### PR TITLE
Fix ModelResponse typing

### PR DESCRIPTION
**Pull request by yujonglee**
This follows https://github.com/BerriAI/litellm/commit/d394c7762368c974d935ab3b97a458d46ee8c361.

### CODE REVIEW

File: litellm/main.py [hunk 1]
```diff
@@ -40,7 +40,7 @@ def completion(
     # model specific optional params
     # used by text-bison only
     top_k=40, request_timeout=0, # unused var for old version of OpenAI API
-  ):
+  ) -> ModelResponse:
```
Comment (yujonglee) [inline litellm/main.py:43]: Seems like type is not inferred here. Add explicit return type.

**Comment by ghost** [comment]
Will have to pull and run this locally - i'll merge this tonight. Thanks for the PR @yujonglee

**Comment by ghost** [comment]
@yujonglee do you have a twitter - appreciate the contribution. Would love to give you a shoutout.

**Comment by ghost** [comment]
Hey, following up on this - is there a **strong** reason why we need this? 

This leads to errors using litellm in a flask app, as the object isn't a dict. 


Screenshot 2023-08-17 at 8 14 17 PM

**Comment by ghost** [comment]
cc: @ishaan-jaff

**Comment by ghost** [comment]
image

fyi - this isn't an issue with the openai response object

**Comment by ghost** [comment]
Screenshot 2023-08-17 at 8 22 07 PM

looks like they inherit from the dict type - so this could be a quick fix for us

**Comment by yujonglee** [comment]
@krrishdholakia 
This changes make autocomplete works.
Screenshot 2023-08-18 at 12 22 50 PM

My handle is `@yujonglee`

Can you give me script you are running `Flask`? I can look into it.

**Comment by ghost** [comment]
Ah - makes sense. I just pushed a fix - inheriting from `dict`, will update this thread on status.

**Comment by ghost** [comment]
@yujonglee i'm testing it on this - https://github.com/BerriAI/litellm/tree/main/cookbook/proxy-server

update: it didn't fix it (v0.1.421)

**Comment by ghost** [comment]
Unable to iterate through the response object and convert it to a dict either (@ishaan-jaff)
code: 
```
    response_dict = {}
    for key in response:
        response_dict[key] = response
```

Error
```
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/flask/app.py", line 2525, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/flask/app.py", line 1823, in full_dispatch_request
    return self.finalize_request(rv)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/flask/app.py", line 1842, in finalize_request
    response = self.make_response(rv)
               ^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/flask/app.py", line 2153, in make_response
    rv = self.json.response(rv)
         ^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/flask/json/provider.py", line 309, in response
    f"{self.dumps(obj, **dump_args)}\n", mimetype=mimetype
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/flask/json/provider.py", line 230, in dumps
    return json.dumps(obj, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/__init__.py", line 238, in dumps
    **kw).encode(obj)
          ^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/encoder.py", line 200, in encode
    chunks = self.iterencode(o, _one_shot=True)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/json/encoder.py", line 258, in iterencode
    return _iterencode(o, 0)
           ^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/flask/json/provider.py", line 122, in _default
    raise TypeError(f"Object of type {type(o).__name__} is not JSON serializable")
TypeError: Object of type ModelResponse is not JSON serializable
```

**Comment by ghost** [comment]
Reverting my local instance to last commit before model types - `0.1.411`

**Comment by ishaan-jaff** [comment]
@yujonglee thanks for this PR it looks awesome !

File: litellm/main.py (hunk 1 above)
Comment (ishaan-jaff) [reply litellm/main.py:43]: for OpenAI we don't return `ModelResponse` , can you add a test in test_completion to ensure this doesn't break openai responses. 

essentially both of these should still work for OpenAI
```python
print(response.choices[0].message.content)
print(response['choices'][0]['message']['content'])
```

**Review by ishaan-jaff — COMMENTED** [review]
otherwise this PR lgtm ! 

If you can verify it does not break OpenAI responses - we can merge it !

**Comment by ishaan-jaff** [comment]
@krrishdholakia let's merge this and then deal with the flask error ?

File: litellm/tests/test_completion.py [hunk 2]
```diff
@@ -94,14 +94,9 @@ def test_completion_openai():
...
-        print(response_str_2)
-        if type(response_str) != str:
-            pytest.fail(f"Error occurred: {e}")
-        if type(response_str_2) != str:
-            pytest.fail(f"Error occurred: {e}")
-        # Add any assertions here to check the response
-        print(response)
+        assert response_str == response_str_2
+        assert type(response_str) == str
+        assert len(response_str) > 1
```
Comment (yujonglee) [inline litellm/tests/test_completion.py:99]: > for OpenAI we don't return `ModelResponse` , can you add a test in test_completion to ensure this doesn't break openai responses.
> 
> essentially both of these should still work for OpenAI
> 
> ```python
> print(response.choices[0].message.content)
> print(response['choices'][0]['message']['content'])
> ```

It seems openai didn't break

**Comment by ghost** [comment]
The issue with flask errors is now fixed. 

version: pip install litellm==0.1.424
commit: https://github.com/BerriAI/litellm/tree/ee20e6a1e63441b51363b03100dfe0fa95b3398c

**Pull request merged**